discovery_engine_core/web-api2/src/error/application.rs

```diff
@@ -105,11 +105,11 @@ macro_rules! impl_application_error {
                 stringify!($name)
             }
 
-            fn encode_details(&self) -> serde_json::Value {
+            fn encode_details(&self) -> ::serde_json::Value {
                 ::serde_json::to_value(self)
                     .unwrap_or_else(|err| {
-                        error!(%err, "serializing error details failed");
-                        serde_json::Value::Null
+                        ::tracing::error!(%err, "serializing error details failed");
```

Comment:
:wrench: the macro is already imported above, no need for a full path here
```suggestion
                        error!(%err, "serializing error details failed");
```

Reply:
The macro is used across the crate, so `error!` might not have been imported.

Reply:
i missed the `#[macro_export]` which was hidden behind github's folding, my bad.